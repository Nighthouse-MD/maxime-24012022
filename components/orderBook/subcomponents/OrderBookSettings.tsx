import React, { memo } from 'react';
import { Row, Col } from 'react-bootstrap';
import styles from '../OrderBook.module.scss'

const OrderBookSettings = ({ depth, handleDepthChange, throttleInterval, handleThrottleIntervalChange }: Props) => {
    return <>
        <Row>
            <Col xs="4" className={styles.settingsLabelCol}>
                Depth
            </Col>
            <Col xs="4" className={styles.settingsInputCol}>
                <input
                    id="depth"
                    className={styles.settingsInput}
                    value={depth}
                    onChange={handleDepthChange}
                    type="range"
                    min="0"
                    max="40"
                />
            </Col>
            <Col xs="4" className={styles.settingsValueCol}>
                {depth}
            </Col>
        </Row>
        <Row>
            <Col xs="4" className={styles.settingsLabelCol}>
                Render throttle
            </Col>
            <Col xs="4" className={styles.settingsInputCol}>
                <input
                    id="renderThrottle"
                    className={styles.settingsInput}
                    value={throttleInterval}
                    onChange={handleThrottleIntervalChange}
                    type="range"
                    min="0"
                    max="2000"
                />
            </Col>
            <Col xs="4" className={styles.settingsValueCol}>
                {throttleInterval} ms
            </Col>
        </Row>
    </>;
}

interface Props {
    depth: number;
    throttleInterval: number;
    handleDepthChange: React.ChangeEventHandler<HTMLInputElement>;
    handleThrottleIntervalChange: React.ChangeEventHandler<HTMLInputElement>;
}

export default memo(OrderBookSettings);