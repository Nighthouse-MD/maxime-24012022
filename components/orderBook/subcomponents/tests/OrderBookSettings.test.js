import OrderBookSettings from '../OrderBookSettings';
import renderer from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react';

describe('OrderBookSettings', () => {
    it('shows all the elements', () => {
        const handleSetDepth = jest.fn();
        const handleSetRenderThrottleInterval = jest.fn();
        render(<OrderBookSettings depth={15} setDepth={handleSetDepth} renderThrottleInterval={350} setRenderThrottleInterval={handleSetRenderThrottleInterval} />);

        expect(screen.queryByText("Depth")).toBeInTheDocument();
        expect(screen.queryByText("Render throttle")).toBeInTheDocument();

        expect(screen.queryByText("15")).toBeInTheDocument();
        expect(screen.queryByText("350 ms")).toBeInTheDocument();

        expect(screen.queryAllByRole('slider').length).toBe(2);
    });

    it('fires the handlers when sliding', () => {
        const handleSetDepth = jest.fn();
        const handleSetRenderThrottleInterval = jest.fn();
        render(<OrderBookSettings depth={15} setDepth={handleSetDepth} renderThrottleInterval={350} setRenderThrottleInterval={handleSetRenderThrottleInterval} />);

        const sliders = screen.queryAllByRole('slider');

        sliders.forEach(slider => fireEvent.change(slider, { target: { value: 20 } }));

        expect(handleSetDepth).toHaveBeenCalledTimes(1);
        expect(handleSetRenderThrottleInterval).toHaveBeenCalledTimes(1);
    });

    it('renders correctly for OrderType.bid', () => {
        const handleSetDepth = jest.fn();
        const handleSetRenderThrottleInterval = jest.fn();

        const tree = renderer
            .create(<OrderBookSettings depth={15} setDepth={handleSetDepth} renderThrottleInterval={350} setRenderThrottleInterval={handleSetRenderThrottleInterval} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});