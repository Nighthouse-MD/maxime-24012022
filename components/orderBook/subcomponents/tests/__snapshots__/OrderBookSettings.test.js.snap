// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderBookSettings renders correctly for OrderType.bid 1`] = `
Array [
  <div
    className="row"
  >
    <div
      className="settingsLabelCol col-4"
    >
      Depth
    </div>
    <div
      className="settingsInputCol col-4"
    >
      <input
        className="settingsInput"
        id="depth"
        max="40"
        min="0"
        onChange={[Function]}
        type="range"
        value={15}
      />
    </div>
    <div
      className="settingsValueCol col-4"
    >
      15
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="settingsLabelCol col-4"
    >
      Render throttle
    </div>
    <div
      className="settingsInputCol col-4"
    >
      <input
        className="settingsInput"
        id="renderThrottle"
        max="2000"
        min="0"
        onChange={[Function]}
        type="range"
        value={350}
      />
    </div>
    <div
      className="settingsValueCol col-4"
    >
      350
       ms
    </div>
  </div>,
]
`;
